### ISSUE TITLE
Request: Support for git worktrees

### ISSUE THREAD

**Issue by Mitsunee** [usmanyunusov/nano-staged#26]
### Clear and concise description of the problem

Currently `git worktree` is not supported. The following error appears:

![Screenshot_20220415_122454](https://user-images.githubusercontent.com/19474909/163560314-5242da8c-16fa-406b-93dd-64036e4003ab.png)

### Suggested solution

before attempting to read `.git` as a directory stat it to determine if it is a worktree (where .git is a file containing a path) or not (where it is a directory)

### Additional context

To recreate create a new project:

```sh
mkdir test-project-master
cd test-project
git init
yarn init -y && yarn add nano-staged prettier
```

Then add a config such as this one to `package.json`

```json
  "nano-staged": {
    "**/*.js": [
      "prettier -w"
    ],
  }
```

Now commit the initial setup, then create a new worktree and attempt to make a commit there

```sh
git add -A
git commit -m "Initial setup"

git worktree add ../test-project-dev -b dev-branch # creates new worktree with its own branch
# note that a branch cannot be checked out by multiple worktrees,
# thus a new branch is required
cd ../test-project-dev
yarn # install in the new worktree

touch a-file.js
git add a-file.js
git commit -m "added a file"
```

You should now get the error as in the current working directory `.git` is a file describing the location of its location in the `.git` directory of the main tree. For more information on worktrees see [git-worktree](https://git-scm.com/docs/git-worktree)

**Comment by usmanyunusov** [comment]
Fixed #27. Release 0.7 version.

**Comment by Mitsunee** [comment]
Just tested it, works great now, thank you :)
